File: 1779.找到最近的有相同X或Y坐标的点/1779-找到最近的有相同X或Y坐标的点.go
func nearestValidPoint(x int, y int, points [][]int) int {
    var ans int = -1
    minDis := math.MaxInt

    for i, p := range points {
        a := p[0]
        b := p[1]
        dis := math.MaxInt
        if a==x {
            dis = abs(b-y)
        } else if b==y {
            dis = abs(a-x)
        }

        if dis < minDis {
            minDis = dis
            ans = i
        }
    }

    return ans
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

File: 1779.找到最近的有相同X或Y坐标的点/1779-找到最近的有相同X或Y坐标的点.cpp
class Solution {
public:
    int nearestValidPoint(int x, int y, vector<vector<int>>& points) {
        int minDst = INT_MAX;
        int ans = -1;
        for( int i=0; i<points.size(); i++ ) {
            int a = points[i][0];
            int b = points[i][1];
            int dst = INT_MAX;
            if (x == a) {
                dst = abs(b - y);
            } else if (y==b) {
                dst = abs(a-x);
            }
            if (dst < minDst) {
                minDst = dst;
                ans = i;
            }
        }

        return ans;
    }
};